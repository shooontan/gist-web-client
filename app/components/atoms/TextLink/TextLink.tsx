import * as React from 'react';
import Link, { LinkProps } from 'next/link';
import pure from 'recompose/pure';
import styled from 'styled-components';
import { Overwrite } from 'typelevel-ts';

type Props = Overwrite<
  React.LinkHTMLAttributes<HTMLLinkElement>,
  {
    href: LinkProps['href'];
  }
>;

const TextLink = (props: Props) => {
  const { children, href, ...rest } = props;
  return (
    <Link href={href} passHref {...rest}>
      <Anker>{children}</Anker>
    </Link>
  );
};

export default pure(TextLink);

const Anker = styled.a`
  text-decoration: none;
`;
