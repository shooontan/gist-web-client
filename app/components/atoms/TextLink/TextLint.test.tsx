import { render } from 'react-testing-library';
import TextLink from './TextLink';

const href = 'https://example.com/';
const hrefUser = {
  pathname: '/user',
  query: { name: 'username' },
};

test('snapshot Icon', () => {
  const { container } = render(<TextLink href={href} />);
  expect(container.firstChild).toMatchSnapshot();

  const { container: textContainer } = render(
    <TextLink href={href}>TextLink Component</TextLink>
  );
  expect(textContainer.firstChild).toMatchSnapshot();

  const { container: paramsHrefContainer } = render(
    <TextLink href={hrefUser}>user name</TextLink>
  );
  expect(paramsHrefContainer.firstChild).toMatchSnapshot();
});
