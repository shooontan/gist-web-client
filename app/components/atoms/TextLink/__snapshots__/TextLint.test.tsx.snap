// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot Icon 1`] = `
.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
}

<a
  class="c0"
  href="https://example.com/"
/>
`;

exports[`snapshot Icon 2`] = `
.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
}

<a
  class="c0"
  href="https://example.com/"
>
  TextLink Component
</a>
`;

exports[`snapshot Icon 3`] = `
.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
}

<a
  class="c0"
  href="/user?name=username"
>
  user name
</a>
`;
